spinalcordtoolbox/reports/qc.py

```diff
@@ -220,8 +236,14 @@ def __init__(self, input_file, command, args, orientation, dest_folder):
         abs_input_path = os.path.dirname(os.path.abspath(input_file))
         abs_input_path, contrast = os.path.split(abs_input_path)
         _, subject = os.path.split(abs_input_path)
+        command = command or os.path.splitext(os.path.basename(sys.argv[0]))[0]
```

Comment:
Consider splitting intermediate contents of this statement as variables. For instance:

    command_name = sys.argv[0]
    command_basename = os.path.basename(command_name)
    # (...)

Then each step will be more meaningful for who is reading.

Reply:
Philosophical question here...

Each line of code has value and a cost, and in this case I believe that saving lines is reducing the cost, and raises the chance of someone who doesn't know about the `os.path` API to go and document herself.
The line is still very simple; for example you can find offenders of several orders of magnitude by doing a `git grep "=.*(.*(.*(.*).*).*)"` in the code.